File: src/main/java/com/github/liebharc/queryenrichment/QueryBuilder.java
package com.github.liebharc.queryenrichment;

import java.util.List;

/**
 * A query builder prepares a {@link Query}.
 */
public interface QueryBuilder {
    /**
     * Builds a query.
     * @param selectors Select expression
     * @param filters FilterStep or where expression
     * @return Query object
     */
    Query build(List<QuerySelector> selectors, List<QueryFilter> filters);
}

File: src/main/java/com/github/liebharc/queryenrichment/QueryBuilder.kt
package com.github.liebharc.queryenrichment

/**
 * A query builder prepares a [Query].
 */
interface QueryBuilder {
    /**
     * Builds a query.
     * @param selectors Select expression
     * @param filters FilterStep or where expression
     * @return Query object
     */
    fun build(selectors: List<QuerySelector>, filters: List<QueryFilter>): Query
}